### PR TITLE
Update doc for version 0.2.0

### LINKED ISSUE #43: Update docs for new config system
PR https://github.com/Omerlo-Technologies/ex_prosemirror/pull/42 breaks a lot of things.

We should rewrite the doc

### PR DESCRIPTION
**Pull request by Matsa59**
## 📖 Description

Update the doc for the version `0.2.0`

Plus Adding some categories

Close #43 

## 📋 Type of change

- [x] This change requires a documentation update

Screen Shot 2021-06-09 at 5 03 55 PM

### CODE REVIEW

File: lib/ex_prosemirror/block/heading.ex [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
-  Create heading text style using `<hX></hX>` html style.
-
-  > `X` is a number between 1 and 6.
-
-  ## inputs opts:
-
-      blocks: [{:heading, [:h1, :h2, :h3, :h4, :h5, :h6]}]
-
+  Heading contains multiples ExProsemirror.Block.Text and have attributes that
+  define the level of the heading (between 1 and 6).
```
Comment (tomkonidas) [inline lib/ex_prosemirror/block/heading.ex:4]:
```suggestion
  Heading contains multiple ExProsemirror.Block.Text and have attributes that
  define the level of the heading (between 1 and 6).
```

File: lib/ex_prosemirror/block/paragraph.ex [hunk 2]
```diff
@@ -1,11 +1,6 @@
 defmodule ExProsemirror.Block.Paragraph do
   @moduledoc ~S"""
-  Create heading text style using `<p></p>` html style.
-
-  ## inputs opts:
-
-      blocks: [:p]
-
+  A paragraph that contains multiples ExProsemirror.Block.Text.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/block/paragraph.ex:3]:
```suggestion
  A paragraph that contains multiple ExProsemirror.Block.Text.
```

File: lib/ex_prosemirror/modifier_helper.ex [hunk 3]
```diff
@@ -1,7 +1,10 @@
-defmodule ExProsemirror.SchemaHelper do
+defmodule ExProsemirror.ModifierHelper do
   @moduledoc ~S"""
   Ecto schema helper that defines multiple macros to use inside an Ecto.Schema.
 
+  In contrast of `ExProsemirror.Schema`, this module should not be importer to your business schema.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/modifier_helper.ex:5]:
```suggestion
  In contrast of `ExProsemirror.Schema`, this module should not be imported to your business schema.
```

File: lib/ex_prosemirror/modifier_helper.ex [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
+  It MUST be use to create custom block / marks.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/modifier_helper.ex:6]:
```suggestion
  It **MUST** be used to create custom block / marks.
```

the double `*` should make it bold, but check just to be sure, not sure the docs will read it like this.

[Reference](https://www.markdownguide.org/basic-syntax#bold)

File: lib/ex_prosemirror/modifier_helper.ex [hunk 5]
```diff
@@ -16,7 +19,11 @@ defmodule ExProsemirror.SchemaHelper do
 
         embedded_prosemirror_content([text: ExProsemirror.Block.Text])
 
-  Use macro `ExProsemirror.SchemaHelper.embedded_prosemirror_field/3`.
+  ## Options
+
+  - array: boolean that define if we could have multiple elements.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/modifier_helper.ex:24]:
```suggestion
  - array: boolean that defines if we could have multiple elements.
```

File: lib/ex_prosemirror/modifier_helper.ex [hunk 6]
```diff
@@ -83,6 +92,26 @@ defmodule ExProsemirror.SchemaHelper do
...
+
+  * %ExProsemirror.Block.Text{...}
+  * attrs (map of data to cast)
+  * opts: opts to use in the changeset (e.g. allowed marks to cast defined in the ExProsemirror type.)
+
+  > If a block/mark's type is not specified in the `with` opts, the `changeset/2` will be use.
+
+  ## Options
+
+  * `with`: define the changeset to apply, to send allowed marks we recommands
```
Comment (tomkonidas) [inline lib/ex_prosemirror/modifier_helper.ex:112]:
```suggestion
  * `with`: define the changeset to apply, to send allowed marks we recommend
```

**Review by tomkonidas — APPROVED** [review]

**Review by onixus74 — APPROVED** [review]

**Pull request merged**